### PR TITLE
Makes secbots AND SLIMES use bayesian logic to determine threat levels

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

See title. Starting with a prior of 5% likely to be a threat, threat determination goes through a bayesian updates on available evidence to determine likelihood of scumbag. It's probably overzealous. Oh well!

## Why It's Good For The Game

Lets you change the sensitivity of secbots, for one (not *that* much, setting it to 0 would make it basically an emag). Also, I think "85.3% chance of scumbag" is a funny phrase.

## Changelog
:cl:
tweak: Secbots now use bayesian statistics (based on priors i pulled out my butt, mind) to determine threat
add: Secbots can have their minimum threat level required to arrest adjusted. Those set to arrest are always arrested, obviously. 
balance: Secbot change leads to unknown knock-on effects that will need to be tweaked.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -591,25 +591,25 @@
...
 
 	//mindshield implants imply trustworthyness
 	if(HAS_TRAIT(src, TRAIT_MINDSHIELD))
-		threatcount -= 1
+		BAYES_THEOREM(threat_perc,0.005,0.05) // mindshielded people HIGHLY unlikely to be scumbags
 
 	//Agent cards lower threatlevel.
 	if(istype(idcard, /obj/item/card/id/syndicate))
-		threatcount -= 2
+		BAYES_THEOREM(threat_perc,0.005,0.05) // fools 'em into thinking there's a mindshield, basically
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/human.dm]: The reason its -2 is so you can mask + carry without sec bots going `Da fuck` I thought

Comment (Putnam3145) [reply code/modules/mob/living/carbon/human/human.dm]: yes, this also does that

**Review by Ghommie — APPROVED** [review]

**Comment by silicons** [comment]
This seems pretty fun but I'm worried it will make them a bit too annoying to configure @Putnam3145

**Comment by Putnam3145** [comment]
might wanna review this again.

and yeah "annoying to configure" is right i kinda did this as a goof? i've even thought about updating the priors round-to-round and silly things like that.

**Comment by silicons** [comment]
requested

**Pull request closed without merging**